src/lib.rs

```diff
@@ -91,4 +91,48 @@ impl StrRange {
     pub fn end(self) -> StrIndex {
         self.end
     }
+
+    /// Do these ranges overlap?
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use str_index::*;
+    /// let left = StrRange::between(0.into(), 20.into());
+    /// let right = StrRange::between(10.into(), 30.into());
+    /// assert!(left.overlaps(right));
+    /// assert!(right.overlaps(left));
+    ///
+    /// let left = StrRange::between(0.into(), 10.into());
+    /// let right = StrRange::between(10.into(), 20.into());
+    /// assert!(!left.overlaps(right));
+    /// assert!(!right.overlaps(left));
+    ///
+    /// let outer = StrRange::between(0.into(), 20.into());
+    /// let inner = StrRange::between(5.into(), 15.into());
+    /// assert!(outer.overlaps(inner));
+    /// assert!(inner.overlaps(outer));
+    /// ```
+    pub fn overlaps(self, other: StrRange) -> bool {
+        self.start() <= other.start() && self.end() > other.start()
+            || other.start() <= self.start() && other.end() > self.start()
+    }
+
+    /// Does this range contain `other`?
+    ///
+    /// `other` must be completely within `self`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use str_index::*;
+    /// let range = StrRange::between(0.into(), 20.into());
+    /// assert!(range.contains(StrRange::between(5.into(), 15.into())));
+    /// assert!(range.contains(StrIndex::from(10)));
+    /// assert!(range.contains(range.end()));
+    /// ```
+    pub fn contains<I: Into<StrRange>>(self, other: I) -> bool {
+        let other = other.into();
+        self.start() <= other.start() && self.end() >= other.end()
```

Comment:
That’s very subjective, but if you only use < and <= and avoid > and >=, comparisons become easier to read because you can visualize them on a number line

Reply:
Note also that I am slightly uneasy about `Into<StrRange>`. Specifically, when testing if an offset is withing a range, you might want `<` or `<=` on the upper bound in different situations. 

Note that IntelliJ has both versions, and they are named almost the same: 

* https://github.com/JetBrains/intellij-community/blob/9475133fbf59b528645aaecdc085d0f224b68fac/platform/util/src/com/intellij/openapi/util/TextRange.java#L102-L104

* https://github.com/JetBrains/intellij-community/blob/9475133fbf59b528645aaecdc085d0f224b68fac/platform/util/src/com/intellij/openapi/util/TextRange.java#L111-L113

That's definitelly something we don't want to do :)

Reply:
Yeah, I'm rethinking that now. I think I'm going to add some more methods to `StrIndex` instead.

Reply:
These are now spelled

- `range.contains(index.as_unit_range())` and
- `range.contains_exclusive(index)`